### PR TITLE
feat: support es compression

### PR DESCRIPTION
**Pull request by jiuker**
## Community Contribution License
All community contributions in this pull request are licensed to the project maintainers
under the terms of the [Apache 2 license](https://www.apache.org/licenses/LICENSE-2.0). 
By creating this pull request I represent that I have the right to license the 
contributions to the project maintainers under the Apache 2 license.

## Description

feat: support es compression

## Types of changes
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Optimization (provides speedup with no functional changes)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:
- [ ] Fixes a regression (If yes, please add `commit-id` or `PR #` here)
- [ ] Unit tests added/updated
- [ ] Internal documentation updated
- [ ] Create a documentation update request [here](https://github.com/minio/docs/issues/new?label=doc-change,title=Doc+Updated+Needed+For+PR+github.com%2fminio%2fminio%2fpull%2fNNNNN)

### CODE REVIEW

File: internal/config/notify/help.go [hunk 1]
```diff
@@ -262,6 +262,18 @@ var (
 			Optional:    true,
 			Type:        "sentence",
 		},
+		config.HelpKV{
+			Key:         target.KafkaCompressionCodec,
+			Description: "specify compression_codec of the kafka",
```
Comment (shtripat) [inline internal/config/notify/help.go]: %s/specify compression_codec of the kafka/specify compression codec of the Kafka cluster/g

File: internal/config/notify/help.go [hunk 2]
```diff
@@ -262,6 +262,18 @@ var (
...
+			Optional:    true,
+			Type:        "none|snappy|gzip|lz4|zstd",
+		},
+		config.HelpKV{
+			Key:         target.KafkaCompressionLevel,
+			Description: "specify compression level of the kafka cluster",
```
Comment (shtripat) [inline internal/config/notify/help.go]: %s/specify compression level of the kafka cluster/specify compression level of the Kafka cluster/g

**Review by shtripat — APPROVED** [review]
LGTM. Couple of minor spelling corrections to align with other descriptions.
Able to set/get new fields for `notify_kafka` as below
```
$ mc admin config set myminio notify_kafka                                                                 
KEY:
notify_kafka[:name]  publish bucket notifications to Kafka endpoints

ARGS:
enable*            (on|off)                     enable notify_kafka target, default is 'off'
brokers*           (csv)                        comma separated list of Kafka broker addresses
topic              (string)                     Kafka topic used for bucket notifications
sasl_username      (string)                     username for SASL/PLAIN or SASL/SCRAM authentication
sasl_password      (string)                     password for SASL/PLAIN or SASL/SCRAM authentication
sasl_mechanism     (string)                     sasl authentication mechanism, default 'plain'
tls_client_auth    (string)                     clientAuth determines the Kafka server's policy for TLS client auth
sasl               (on|off)                     set to 'on' to enable SASL authentication
tls                (on|off)                     set to 'on' to enable TLS
tls_skip_verify    (on|off)                     trust server TLS without verification, defaults to "on" (verify)
client_tls_cert    (path)                       path to client certificate for mTLS auth
client_tls_key     (path)                       path to client key for mTLS auth
queue_dir          (path)                       staging dir for undelivered messages e.g. '/home/events'
queue_limit        (number)                     maximum limit for undelivered messages, defaults to '100000'
version            (string)                     specify the version of the Kafka cluster
comment            (sentence)                   optionally add a comment to this setting
compression_codec  (none|snappy|gzip|lz4|zstd)  specify compression_codec of the kafka
compression_level  (number)                     specify compression level of the kafka cluster

$ mc admin config set myminio notify_kafka enable=on brokers=localhost:9092 topic=quickstart-events compression_codec=gzip compression_level=1
Successfully applied new settings.
Please restart your server 'mc admin service restart myminio'.

$ mc admin service restart myminio                                                                          Restart command successfully sent to `myminio`. Type Ctrl-C to quit or wait to follow the status of the restart process.
...
Restarted `myminio` successfully in 1 seconds

$ mc admin config get myminio notify_kafka
notify_kafka topic=quickstart-events brokers=localhost:9092 sasl_username= sasl_password= sasl_mechanism=plain client_tls_cert= client_tls_key= tls_client_auth=0 sasl=off tls=off tls_skip_verify=off queue_limit=0 queue_dir= version= batch_size=0 compression_codec=gzip compression_level=1 
```

File: internal/event/target/kafka_scram_client_contrib.go [hunk 3]
```diff
@@ -38,6 +38,20 @@ func initScramClient(args KafkaArgs, config *sarama.Config) {
 		// default to PLAIN
 		config.Net.SASL.Mechanism = sarama.SASLMechanism(sarama.SASLTypePlaintext)
 	}
+	// set producer
+	codecs := map[string]sarama.CompressionCodec{
```
Comment (Praveenrajmani) [inline internal/event/target/kafka_scram_client_contrib.go]: can we define this as a global var on top of the file?

File: internal/event/target/kafka_scram_client_contrib.go [hunk 4]
```diff
@@ -38,6 +38,20 @@ func initScramClient(args KafkaArgs, config *sarama.Config) {
...
+		"snappy": sarama.CompressionSnappy,
+		"lz4":    sarama.CompressionLZ4,
+		"zstd":   sarama.CompressionZSTD,
+	}
+	cc, ok := codecs[strings.ToLower(args.Producer.Compression)]
+	if !ok {
+		return
+	}
+	config.Producer.Compression = cc
+	config.Producer.CompressionLevel = args.Producer.CompressionLevel
```
Comment (Praveenrajmani) [inline internal/event/target/kafka_scram_client_contrib.go]: can we move this setting outside of `initScramClient` function? we can have it in `initKafka()` instead

File: docs/bucket/notifications/README.md [hunk 5]
```diff
@@ -1134,22 +1134,24 @@ KEY:
...
+MINIO_NOTIFY_KAFKA_TLS                         (on|off)                    set to 'on' to enable TLS
+MINIO_NOTIFY_KAFKA_TLS_SKIP_VERIFY             (on|off)                    trust server TLS without verification, defaults to "on" (verify)
+MINIO_NOTIFY_KAFKA_CLIENT_TLS_CERT             (path)                      path to client certificate for mTLS auth
+MINIO_NOTIFY_KAFKA_CLIENT_TLS_KEY              (path)                      path to client key for mTLS auth
+MINIO_NOTIFY_KAFKA_QUEUE_DIR                   (path)                      staging dir for undelivered messages e.g. '/home/events'
+MINIO_NOTIFY_KAFKA_QUEUE_LIMIT                 (number)                    maximum limit for undelivered messages, defaults to '100000'
+MINIO_NOTIFY_KAFKA_COMMENT                     (sentence)                  optionally add a comment to this setting
+MINIO_NOTIFY_KAFKA_VERSION                     (string)                    specify the version of the Kafka cluster e.g. '2.2.0'
+MINIO_NOTIFY_KAFKA_PRODUCER_COMPRESSION_CODEC  (none|snappy|gzip|lz4|zstd) compression codec for producer messages, defaults to 'none'
```
Comment (Praveenrajmani) [inline docs/bucket/notifications/README.md]:
```suggestion
MINIO_NOTIFY_KAFKA_PRODUCER_COMPRESSION_CODEC  (none|snappy|gzip|lz4|zstd) compression codec for producer messages
```

I see that we don't have any defaults here.

**Review by Praveenrajmani — COMMENTED** [review]
few minor nits, lgtm otherwise.

**Review by shtripat — APPROVED** [review]

**Review by Praveenrajmani — APPROVED** [review]

**Review by harshavardhana — APPROVED** [review]

**Pull request merged**